src/canvas-extractor/tests/graphql/conftest.py

```diff
@@ -34,19 +47,33 @@ def gql(api):
         START_DATE,
         END_DATE,
         )
-    gql.extract(api)
+    gql.extract(read_data)
+
     yield gql
 
 
-@pytest.fixture(scope="session")
-def api():
-    '''
-    Returns the data converted from json file
-    '''
-    with open('tests/graphql/sample-data.json', mode='r') as f:
-        data = json.loads(f.read())
+@pytest.fixture(scope="class")
+def gql():
+    CANVAS_BASE_URL = os.environ['CANVAS_BASE_URL']
+    CANVAS_ACCESS_TOKEN = os.environ['CANVAS_ACCESS_TOKEN']
+    START_DATE = "2021-01-01"
+    END_DATE = "2030-01-01"
 
-    return data
+    gql = GraphQLExtractor(
+        CANVAS_BASE_URL,
+        CANVAS_ACCESS_TOKEN,
+        "1",
+        START_DATE,
+        END_DATE,
+        )
+    gql.run()
+
+    yield gql
+
+
+@pytest.fixture(autouse=True, scope="class")
+def load_env():
+    load_dotenv()
```

Comment:
SHOULD: `load_dotenv()` can be directly called without wrapping it as a fixture.